packages/modules/inventory/src/models/inventory-item.ts

```diff
@@ -1,156 +1,46 @@
-import {
-  createPsqlIndexStatementHelper,
-  DALUtils,
-  generateEntityId,
-  Searchable,
-} from "@medusajs/framework/utils"
-import {
-  BeforeCreate,
-  Collection,
-  Entity,
-  Filter,
-  Formula,
-  OneToMany,
-  OnInit,
-  OptionalProps,
-  PrimaryKey,
-  Property,
-  Rel,
-} from "@mikro-orm/core"
-
-import { DAL } from "@medusajs/framework/types"
-import { InventoryLevel } from "./inventory-level"
-import { ReservationItem } from "./reservation-item"
-
-const InventoryItemDeletedAtIndex = createPsqlIndexStatementHelper({
-  tableName: "inventory_item",
-  columns: "deleted_at",
-  where: "deleted_at IS NOT NULL",
-})
-
-const InventoryItemSkuIndex = createPsqlIndexStatementHelper({
-  tableName: "inventory_item",
-  columns: "sku",
-  unique: true,
-  where: "deleted_at IS NULL",
-})
-
-type InventoryItemOptionalProps = DAL.SoftDeletableModelDateColumns
-
-@Entity()
-@Filter(DALUtils.mikroOrmSoftDeletableFilterOptions)
-export class InventoryItem {
-  [OptionalProps]: InventoryItemOptionalProps
-
-  @PrimaryKey({ columnType: "text" })
-  id: string
-
-  @Property({
-    onCreate: () => new Date(),
-    columnType: "timestamptz",
-    defaultRaw: "now()",
+import { model } from "@medusajs/framework/utils"
+import InventoryLevel from "./inventory-level"
+import ReservationItem from "./reservation-item"
+
+const InventoryItem = model
+  .define("InventoryItem", {
+    id: model.id({ prefix: "iitem" }).primaryKey(),
+    sku: model.text().searchable().nullable(),
+    origin_country: model.text().nullable(),
+    hs_code: model.text().searchable().nullable(),
+    mid_code: model.text().searchable().nullable(),
+    material: model.text().nullable(),
+    weight: model.number().nullable(),
+    length: model.number().nullable(),
+    height: model.number().nullable(),
+    width: model.number().nullable(),
+    requires_shipping: model.boolean().default(true),
+    description: model.text().searchable().nullable(),
+    title: model.text().searchable().nullable(),
+    thumbnail: model.text().nullable(),
+    metadata: model.json().nullable(),
+    location_levels: model.hasMany(() => InventoryLevel, {
+      mappedBy: "inventory_item",
+    }),
+    reservation_items: model.hasMany(() => ReservationItem, {
+      mappedBy: "inventory_item",
+    }),
+    reserved_quantity: model.number().nullable(), // .readonly?
+    stocked_quantity: model.number().nullable(), // .readonly?
```

Comment:
create `.readonly` modifier or similar?

Reply:
@olivermrbl ideally we need a modifier to flag the computed fields here. let me know if we should add it.

Reply:
Yeah, I think that's a good idea – can we do `.computed()` maybe?